codegen/src/types/composite_def.rs

```diff
@@ -41,6 +42,7 @@ pub struct CompositeDef {
 
 impl CompositeDef {
     /// Construct a definition which will generate code for a standalone `struct`.
+    #[allow(clippy::too_many_arguments)]
```

Comment:
Possibly create a follow-up ticket to refactor with a config struct instead of the params.